thunder/tests/test_grad.py

```diff
@@ -1700,6 +1700,82 @@ def func(a, b):
         get_saved_for_backward_tensors(execution_trace)
 
 
+def test_torch_checkpoint():
+    import torch.utils.checkpoint
+    import torch._higher_order_ops.wrap
+
+    def fn_to_checkpoint(x):
+        return x.sin().cos().exp()
```

Comment:
When this PR is ready the test should be working with this
```suggestion
        return torch.exp(torch.cos(torch.sin(x)))
```